### DISCUSSION TITLE
Moose way of representing units for various quantities

### DISCUSSION THREAD

**Discussion by abarun22** [idaholab/moose#25075] [Q&A General]
Hi all,
Just thought of clarifying my ambiguities associated with addressing units in moose for various quantities used in the computation. I am aware these are revolving mainly around the SI units as can be seen [here](https://mooseframework.inl.gov/source/utils/Units.html#:~:text=Upon%20parsing%20all%20units%20are,units%20can%20be%20converted%20between).). 
Going by that, is it a valid statement that i address stresses by Pa, assuming that the length scale moose uses is 'meter' and force in kgm/s^2.

The main intention behind my question being that i am struggling to find the units for some of the constants mentioned in https://mooseframework.inl.gov/source/materials/crystal_plasticity/CrystalPlasticityKalidindiUpdate.html#!, but unable to find descriptions in the parameters section. The t_sat in that case would be literally called as 109.8 Pa if i am right.
Kind regards,
Arun

**Comment by sapitts** [comment]
Hi @abarun22,
The crystal plasticity classes deviate from the unit-agnostic approach taken by default in MOOSE. As described under the section **Units Assumed in Crystal Plasticity Materials** on the [Compute Multiple Crystal Plasticity Stress documentation page](https://mooseframework.inl.gov/source/materials/crystal_plasticity/ComputeMultipleCrystalPlasticityStress.html), the crystal plasticity models use the mm-MPa-s unit system. In addition to the documentation page, we have tried to explicitly list the assumed units in the parameter description. 

It does look like the Kalidindi model you are referencing does not list those unit assumptions in the parameter descriptions. While we work on updating that documentation, please know that that units for that class is also in the mm-MPa-s system.

Hope this helps,
Stephanie

**Reply by abarun22** [reply]
Thanks Stephanie. That's really what i am after.
Best regards,
Arun
